HeinOnline.js

```diff
@@ -0,0 +1,224 @@
+{
+	"translatorID": "3dcbb947-f7e3-4bbd-a4e5-717f3701d624",
+	"label": "HeinOnline",
+	"creator": "Frank Bennett",
+	"target": "^https?://(www\\.)?heinonline\\.org/HOL/(?:LuceneSearch|Page|IFLPMetaData)\\?",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsbv",
+	"lastUpdated": "2016-02-22 23:08:21"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	Copyright © 2015-2016 Frank Bennett
+	This file is part of Zotero.
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+
+/*
+	***************
+	** Utilities **
+	***************
+*/
+
+// Get any search results from current page
+// Used in detectWeb() and doWeb()
+function getSearchResults(doc) {
+	var results = doc.getElementsByClassName("lucene_search_result_b"),
+		items = {},
+		found = false;
+	for (var i=0, ilen=results.length; i<ilen; i++) {
+		var url = getXPathStr("href", results[i], './/a[1]');
+
+		var title = getXPathStr("textContent", results[i], './/a[1]');
+		title = ZU.trimInternal(title);
+		//title = title.replace(/\s*\[[^\]]*\]$/, '');
+
+		if (!title || !url) continue;
+		
+		items[url] = title;
+		found = true;
+	}
+	return found ? items : false;
+}
+
+// Get the string value of the first object matching XPath
+function getXPathStr(attr, elem, path) {
+	var res = ZU.xpath(elem, path);
+	res = res.length ? res[0][attr] : '';
+	return res ? res : '';
+}
+
+// Extract query values to keys on an object
+function extractQueryValues(url) {
+	var ret = {};
+	ret.base = url.replace(/(.*?)[a-zA-Z]+\?.*/, "$1");
```

Comment:
I get that, but when would that ever be necessary? Since there's only one ?, the non-greedy match has no effect. Maybe I'm just missing an example, but I'm thinking of a URL like
http://heinonline.org/HOL/Page?<somequerystring>  url.replace(/[a-zA-Z]+\?.*/, ""); chops of everything from Page? and returns the rest and your version keeps everything until HOL/ and returns it. In other words, the two are functionally equivalent, the 2nd one is less efficientl.